### DISCUSSION TITLE
I fixed devtools addon for MapWriter.writeMapAsXml() error

### DISCUSSION THREAD

**Discussion by HaaiHenkie** [freeplane/freeplane#1793] [General]
Hello,

I fixed an error with MapWriter.writeMapAsXml() in releaseAddon.groovy. I actually copied the solution from @EdoFro 's fork. Thanks for that.!

I also fixed a problem in checkAddon.groovy when the addon build mind map is in the Gradle plugin directory structure. If the addon build mind map is in src/addon it will check for scripts in src/scripts and subdirectories.

I have submitted a pull request: https://github.com/freeplane/addons/pull/13

**Comment by HaaiHenkie** [comment]
Hi @dpolivaev, I merged my PR in the freeplane/addons repository. I created a [draft release](https://github.com/freeplane/addons/releases) to attach devtools-v0.9.28.addon.mm and version.properties, since Github does not allow to attach these file types to a message like this. Could you publish them on Sourceforge? Or is that also something I could do myself? In that case you need to add me to the SourceForge project. My SourceForge user name is [haaihenkie](https://sourceforge.net/u/haaihenkie/profile/).

**Reply by dpolivaev** [reply]
Thanks for remembering me, I have added you to the source project as a developer. Can you please check if you have sufficient rights for the release? Please be careful not to occasionally remove the Freeplane releases lol.

**Reply by HaaiHenkie** [reply]
Hi @dpolivaev, yes I am a project member now, Thanks! Unfortunately I cannot upload anything to Files. You need to add me as a Release Technician.

![image](https://github.com/freeplane/freeplane/assets/8525411/a6142732-dfa0-4d75-b9bc-91018818f613)

**Reply by dpolivaev** [reply]
done

**Reply by HaaiHenkie** [reply]
Done as well! I will also update [3.3. Extend with add-ons](https://docs.freeplane.org/getting-started/Add-ons_(install).html)

![image](https://github.com/freeplane/freeplane/assets/8525411/e7056ac8-497a-4b3f-acc4-cb2dc794469a)
